### DISCUSSION TITLE
Getting AssertionError when running the script

### DISCUSSION THREAD

**Discussion by Jino1219** [shivamguys/irctc-cypress-automation#54] [Q&A] [answered]
Timed out retrying after 120000ms: Expected to find content: 'Wed, 31 Jul' within the element: <div.col-xs-12.clearfix> but never did.

287 | 
  288 |                         cy.wrap(div).contains(TRAIN_COACH).click()
> 289 |                         cy.get(`:nth-child(${index + 1}) > .bull-back > app-train-avl-enq > :nth-child(1) > :nth-child(7) > :nth-child(1)`).contains(formatDate(TRAVEL_DATE)).click()
      |                                                                                                                                             ^
  290 |                         cy.get(`:nth-child(${index + 1}) > .bull-back > app-train-avl-enq > [style="padding-top: 10px; padding-bottom: 20px;"]`).contains('Book Now').click()
  291 |                         BOOK_UNTIL_TATKAL_OPENS(div, TRAIN_COACH, TRAVEL_DATE, TRAIN_NO, TATKAL)
  292 | 

Getting the above error on the mentioned below screen and remaining entire script got failed 

<img width="944" alt="Screenshot 2024-07-08 204249" src="https://github.com/shivamguys/irctc-cypress-automation/assets/143801108/4580913b-f411-41db-bab8-272c73446441">

Can anyone please looking to this issue and get them resolved please.

**Comment by Jino1219** [comment]
1) Tatkal Booking Begins......


  0 passing (3m)
  1 failing

  1) IRCTC TATKAL BOOKING
       Tatkal Booking Begins......:
     AssertionError: Timed out retrying after 120000ms: Expected to find content: 'Wed, 31 Jul' within the element: <div.col-xs-12.clearfix> but never did.
      at Context.eval (webpack://irctc-cypress-automation/./cypress/support/commands.js:289:140)

**Comment by Jino1219** [comment]
@shivamguys @drtechie  kindly help me to get this resolved please

**Comment by dhivalogu** [comment, marked as answer]
In commands.js file,

Line number 254, 255, 289, 290

Change "index + 1" to "index + 2"

**Reply by hanisntsolo** [reply]
Will this work?

**Reply by hanisntsolo** [reply]
> In commands.js file,
> 
> 
> 
> Line number 254, 255, 289, 290
> 
> 
> 
> Change "index + 1" to "index + 2"

And why this happened in the first place?

**Reply by dhivalogu** [reply]
Yep it worked. Due to addition of another div within the expected scope. Increasing the index to find the right one.

**Reply by hanisntsolo** [reply]
@shivamguys any comment! Are we bringing this change globally?

**Reply by hanisntsolo** [reply]
@dhivalogu pls raise a pull request.

**Reply by shivamguys** [reply]
@hanisntsolo  This has been addressed and Last Transaction Status popping out has also been fixed  
Check the Cypress test video of the latest code 
[watch latest test run video here](https://cloud.cypress.io/projects/7afdkj/runs/520/test-results/de68ceb2-2890-4dd8-a5a9-c0daeb7d7e89/replay?actions=%5B%5D&browsers=%5B%5D&groups=%5B%5D&isFlaky=%5B%5D&modificationDateRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D&orderBy=EXECUTION_ORDER&oses=%5B%5D&specs=%5B%5D&statuses=%5B%5D&testingTypesEnum=%5B%5D&ts=1720697050168.9473&att=1)

**Comment by Jino1219** [comment]
Issue got resolved. thanks a lot for your help @hanisntsolo @dhivalogu

**Reply by shivamguys** [reply]
Fixed #54  
Fixed #49 
Fixed #51 
Fixed #53
 this as well as **the Last Transaction Status Popping out on screen** .....

# Do Checkout the latest code
